### DISCUSSION TITLE
Need some help with a form

### DISCUSSION THREAD

**Discussion by VincvanAs** [filamentphp/filament#5973] [Help] [answered]
Hello Guys,

So far i'm getting done building the form i need. But there is one thing i can't figure out. Maybe it's because all the time spent on the other issues, i can't oversee it anymore haha.

I have a Model eg. MainModel()

In the form i have a CheckboxList:

                        Forms\Components\CheckboxList::make('second_model')
                            ->multiple()
                            ->options(SecondModel::Active()->pluck('name', 'id'))

At this point i'm getting all the active checkboxes from the SecondModel.

In the SecondModel there is also a 'category' integer field.  So for example:

SecondModel::active()->get() ->
      {
        id: 1,
        name: "1th Category",
        is_active: 1,
        **group: 1,**
      },
      {
        id: 2,
        name: "2nd Category",
        is_active: 1,
        **group: 1,**
      },
      {
        id: 3,
        name: "3th Category",
        is_active: 1,
        **group: 1,**
      },
      {
        id: 4,
        name: "4th Category",
        is_active: 1,
        **group: 2,**
      },
      {
        id: 5,
        name: 5th Category",
        is_active: 1,
        **group: 2,**
      },

I want the checkboxlist to make sure only one option can be checked within a group. 

Can anyone assist me with this or give me a point in the right direction to do this the right way?

Thank you very much!

**Comment by danharrin** [comment, marked as answer]
I think you'd have to write a custom validation rule for this, there's no way to prevent specific combinations of checkboxes in Filament. Unless you define dynamic `options()` that change based on which boxes are already checked, that might also work but could be confusing to users.
